ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -169,32 +279,598 @@ module Json = struct
     | _ ->
         [("event", `Null); ("session", `Null)]
 
+  let of_result obj msg =
+    match msg.msg_result with
+    | None ->
+        `Null
+    | Some (t, _d) ->
+        if obj.name = "event" && String.lowercase_ascii msg.msg_name = "from"
+        then
+          `O
+            [
+              ("type", `String "EventBatch")
+            ; ("func_partial_type", `String "EventBatch")
+            ]
+        else
+          let t', _ = string_of_ty_with_enums t ~action:Deserialize () in
+          `O
+            [
+              ("type", `String t')
+            ; ("func_partial_type", `String (get_fp_type t))
+            ]
+
+  let group_params params =
+    List.fold_left
+      (fun groups (param_release, info) ->
+        match groups with
+        | [] ->
+            [(param_release, [info])]
+        | group :: tl -> (
+          match group with
+          | pr, infos when pr = param_release ->
+              (pr, info :: infos) :: tl
+          | _ ->
+              (param_release, [info]) :: groups
+        )
+      )
+      [] params
+
+  let scan_left func init list =
```

Comment:
What is this doing? Is this not some kind of fold?

Reply:
Aha. We should keep the result that each step folded. For example, if the input:
`[group1; group2; group3]` 
we want to get a result:
`[group1; group1 @ group2; group1 @ group2 @ group3]`

Reply:
These related code has been split to the following PR which will raised after this one got merged.